gapic/src/main/com/google/gapid/perfetto/models/CounterTrack.java

```diff
@@ -99,13 +104,15 @@ private String viewSql() {
         return Data.empty(req);
       }
 
-      Data data = new Data(req, new long[rows + 1], new double[rows + 1]);
+      Data data = new Data(req, new long[rows + 1], new double[rows + 1], new long[rows + 1]);
       res.forEachRow((i, r) -> {
         data.ts[i] = r.getLong(0);
         data.values[i] = r.getDouble(2);
+        data.ids[i] = r.getLong(3);
       });
       data.ts[rows] = res.getLong(rows - 1, 1, 0);
       data.values[rows] = data.values[rows - 1];
+      data.ids[rows] = data.ids[rows - 1];
```

Comment:
Is it possible to add a comment to why we are adding an extra index everywhere? Would help if any one of the functions had a comment.  Ignore this if its too trivial.

Reply:
It's actually for transforming (ts, ts + dur, value) scheme to (ts, value) scheme. The added row is for the (very last timestamp - very last value) pair. 

Since this comes from the design of CounterTrack at the very beginning and done by Pascal, I might not be the best person to explain it. Will omit the comment for this PR, maybe we can re-check this thing when Pascal is back.